admin/resources/js/components/Table.tsx

```diff
@@ -41,27 +106,101 @@ function Table<T extends Record<string, unknown>>({
         ) : null}
         {headerGroups.map((headerGroup) => (
           <tr {...headerGroup.getHeaderGroupProps()}>
-            {headerGroup.headers.map((column) => (
-              <th
-                {...column.getHeaderProps(column.getSortByToggleProps())}
-                key={column.id}
+            {headerGroup.headers.map(
+              (column) =>
+                shouldBeVisible(column.Header) && (
+                  <th
+                    {...column.getHeaderProps(column.getSortByToggleProps())}
+                    key={column.id}
+                  >
+                    {column.render("Header")}
+                    <span>
+                      {column.isSorted && (column.isSortedDesc ? " ▼" : " ▲")}
+                    </span>
+                  </th>
+                ),
+            )}
+            <th>
+              <button
+                style={{
+                  backgroundColor: "transparent",
+                  border: "0",
+                  boxShadow: "none",
+                  marginLeft: "15px",
+                  marginTop: "7px",
+                  cursor: "pointer",
+                }}
+                type="button"
+                onClick={() => {
+                  setShowList(!showList);
+                }}
               >
-                {column.render("Header")}
-                <span>
-                  {column.isSorted && (column.isSortedDesc ? " ▼" : " ▲")}
-                </span>
-              </th>
-            ))}
+                <img
+                  src={SettingsIcon}
+                  alt="settings icon"
+                  style={{
+                    width: "25px",
+                    height: "25",
+                    display: "inline-block",
+                  }}
+                />
+                {showList && (
+                  <ul
```

Comment:
I don't think the checklist should be a child of the gear button - that seems weird semantically. And the settings should remain visible, allowing you to click multiple checkboxes, until you click outside of the checklist.